File: src/main/java/ru/tanaxxt/currencysystem/requests/CurrencyRequest.java
package ru.tanaxxt.currencysystem.requests;

import jakarta.validation.constraints.NotBlank;
import lombok.Getter;
import lombok.NoArgsConstructor;


@Getter
@NoArgsConstructor
public class CurrencyRequest {
    @NotBlank(message = "Name is required")
    private String name;
    @NotBlank(message = "Base currency is required")
    private String baseCurrency = "RUB";
    @NotBlank(message = "Price change range is required")
    private String priceChangeRange;
    private String description;

}

File: src/main/java/ru/tanaxxt/currencysystem/models/CurrencyRequest.kt
package ru.tanaxxt.currencysystem.models

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.NotBlank
import ru.tanaxxt.currencysystem.entities.Currency

data class CurrencyRequest(
    @JsonProperty(value = "name", required = true)
    @field:NotBlank(message = "Name is required")
    val name: String,

    @JsonProperty(value = "baseCurrency", required = true)
    @field:NotBlank(message = "Base currency is required")
    val baseCurrency: String,

    @JsonProperty(value = "priceChangeRange", required = true)
    @field:NotBlank(message = "Price change range is required")
    val priceChangeRange: String,

    @JsonProperty("description")
    val description: String?

) {
    fun toCurrency(): Currency {
        val currency = Currency(
            name = name,
            baseCurrency = baseCurrency,
            priceChangeRange = priceChangeRange,
            description = description
        )

        return currency
    }
}